packages/core/src/tools/ripGrep.test.ts

```diff
@@ -135,9 +213,39 @@ describe('ensureRgPath', () => {
     (downloadRipGrep as Mock).mockRejectedValue(error);
 
     await expect(ensureRgPath()).rejects.toThrow(error);
-    expect(fileExists).toHaveBeenCalledTimes(1);
     expect(downloadRipGrep).toHaveBeenCalledWith('/mock/bin/dir');
   });
+
+  it.runIf(process.platform === 'win32')(
+    'should detect ripgrep when only rg.exe exists on Windows',
+    async () => {
+      const windowsBinDir = String.raw`C:\mock\bin\dir`;
+      const storageBinDirMock = Storage.getGlobalBinDir as Mock;
+      const defaultBinDir = '/mock/bin/dir';
+
+      storageBinDirMock.mockReturnValue(windowsBinDir);
+      (fileExists as Mock).mockImplementation(async (filePath: string) =>
```

Comment:
don't mock the file system. specify a temp bin dir location instead and use the real fs.

Reply:
done